src/AWS.Deploy.ServerMode.Client/Utilities/ServerModeUtilities.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using Amazon.Runtime;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace AWS.Deploy.ServerMode.Client.Utilities
+{
+    public static class ServerModeUtilities
+    {
+        /// <summary>
+        /// Checks server mode health API and waits until the API returns a `Ready` status.
+        /// This is useful when initializing a server mode connection to make sure server mode is ready to accept requests.
+        /// </summary>
+        public static async Task WaitUntilServerModeReady(this RestAPIClient restApiClient, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await WaitUntilHelper.WaitUntil(async () =>
+            {
+                var status = SystemStatus.Error;
+                try
+                {
+                    status = (await restApiClient.HealthAsync()).Status;
+                }
+                catch (Exception)
+                {
+                }
+
+                return status == SystemStatus.Ready;
+            }, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(10), cancellationToken);
+        }
+
+        /// <summary>
+        /// Uses AWS .NET SDK <see cref="FallbackCredentialsFactory"/> to resolve credentials.
+        /// </summary>
+        public static Task<AWSCredentials> ResolveCredentials()
```

Comment:
Now that this method has moved from test code to the product code I'm a little nervous about it being misused thinking it would resolve credentials based on the `--profile` switch or settings file. Can we rename the method to `ResolveDefaultCredentials` and be sure the docs are clear what it is doing any when you should call it?

Reply:
Done